src/screens/components/SubmitForm.js

```diff
@@ -14,85 +19,218 @@ const StyledContainer = styled.div`
   }
 `
 
-const defaultInputStyle = Object.freeze({ marginBottom: 14 })
+const fadeOut = keyframes`
+  from {
+    opacity: 1;
+  }
+
+  to {
+    opacity: 0;
+  }
+`
+
+const StyledMessage = styled.div`
+  color: #fff;
+  margin: 20px auto;
+  text-align: center;
+  padding: 10px 0;
+  border-radius: 5px;
+  opacity: 1;
+  animation: ${fadeOut} 1s linear 2s;
+`
+
+const StyledSuccessMessage = styled(StyledMessage)`
+  background-color: #66d566;
+`
+
+const StyledErrorMessage = styled(StyledMessage)`
+  background-color: #fa7b7b;
+`
+
+const REQUIRED_MESSAGE = 'This field is required'
+const VALID_EMAIL_MESSAGE = 'Please type a valid email'
+const VALID_NUMBER_MESSAGE = 'Please type a valid number'
+const VALID_LIMIT_NUMBER_MESSAGE = 'The number must be greater or equal to zero'
+const MESSAGE_TIME = 3000
+
+const initialValues = {
+  firstName: '',
+  lastName: '',
+  weddingRole: '',
+  email: '',
+  phone: '',
+  destination: '',
+  travelDates: '',
+  numberGuests: '',
+  other: '',
+}
+
+const validationSchema = yup.object().shape({
+  firstName: yup.string().required(REQUIRED_MESSAGE),
+  lastName: yup.string().required(REQUIRED_MESSAGE),
+  weddingRole: yup.string().required(REQUIRED_MESSAGE),
+  email: yup.string().email(VALID_EMAIL_MESSAGE).required(REQUIRED_MESSAGE),
+  phone: yup.string().required(REQUIRED_MESSAGE),
+  destination: yup.string().required(REQUIRED_MESSAGE),
+  travelDates: yup.string().required(REQUIRED_MESSAGE),
+  numberGuests: yup.number()
+    .min(0, VALID_LIMIT_NUMBER_MESSAGE)
+    .typeError(VALID_NUMBER_MESSAGE)
+    .required(REQUIRED_MESSAGE),
+  other: yup.string(),
+})
+
+const defaultInputStyle = Object.freeze({ marginBottom: 10 })
 
 const SubmitForm = () => {
-  const [firstName, setFirstName] = useState('')
-  const [lastName, setLastName] = useState('')
-  const [weddingRole, setWeddingRole] = useState('')
-  const [email, setEmail] = useState('')
-  const [phone, setPhone] = useState('')
-  const [destination, setDestination] = useState('')
-  const [travelDates, setTravelDates] = useState('')
-  const [numberGuests, setNumberGuests] = useState('')
-  const [other, setOther] = useState('')
+  const [successMessageVisible, setSuccessMessageVisible] = useState(false)
+  const [errorMessageVisible, setErrorMessageVisible] = useState(false)
+
+  const showSuccessMessage = () => {
+    setErrorMessageVisible(false)
+    setSuccessMessageVisible(true)
+
+    setTimeout(() => {
```

Comment:
Do you think we could handle the form submit with a `loading` state instead of using a `setTimeout`?

Reply:
In this case the `setTimeout` is used to hide the error and success messages. But I agree with you that it should have a `loading` state.

Reply:
cool, sounds good!